lib/puppet/provider/exec.rb

```diff
@@ -47,7 +47,11 @@ def run(command, check = false)
     # This is backwards compatible all the way to Ruby 1.8.7.
     Timeout::timeout(resource[:timeout], Timeout::Error) do
       cwd = resource[:cwd]
-      cwd ||= Dir.pwd
+      # It's ok if cwd is nil. In that case Puppet::Util::Execution.execute() simply will not attempt to
+      # change the working directory, which is exactly the right behavior when no cwd parameter is
+      # expressed on the resource.  Moreover, attempting to change to the directory that is already
+      # the working directory can fail under some circumstances, so avoiding the directory change attempt
+      # is preferable to defaulting cwd to that directory.
```

Comment:
Commit https://github.com/puppetlabs/puppet/commit/1b74e8e493abe1951f1e535e752521326a63c6d3#diff-c6ae6d3864811d9a6124609b8680e6d2L75-R78 changed the `exec` resource to default to the current working directory, I think because the code for executing the command was moved to a `Dir.chdir` block.